src/mds/MDCache.cc

```diff
@@ -12781,15 +12781,34 @@ int MDCache::dump_cache(std::string_view fn, Formatter *f)
     return 1;
   };
 
+  auto start = mono_clock::now();
+  int64_t count = 0;
   for (auto &p : inode_map) {
     r = dump_func(p.second);
     if (r < 0)
       goto out;
+    if (!(++count % 1000) &&
+	timeout > 0 &&
+	std::chrono::duration<double>(mono_clock::now() - start).count() > timeout) {
+      f->open_object_section("abort");
+      f->dump_string("error", "the operation timeout");
+      f->close_section();
+      goto out;
+    }
   }
   for (auto &p : snap_inode_map) {
     r = dump_func(p.second);
     if (r < 0)
       goto out;
+    if (!(++count % 1000) &&
+		timeout > 0 &&
+	std::chrono::duration<double>(mono_clock::now() - start).count() > timeout) {
+      f->open_object_section("abort");
```

Comment:
"inodes" is an array so this will just append an object to the end of the inodes array I believe.

Let's instead modify `dump cache` to output an object:

````
{
   "result": {
   },
   "inodes": [
   ],
}
````

We actually sorta do that already when we detect the cache is too large but the output type should not change. Can you clean that up?

Reply:
I will update it later. thx